core/src/test/java/org/elasticsearch/indices/IndicesServiceTests.java

```diff
@@ -187,4 +199,54 @@ public void testPendingTasks() throws Exception {
         assertAcked(client().admin().indices().prepareOpen("test"));
 
     }
+
+    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/17695")
+    public void testDeletingSameIndexDirectoryFromConcurrentProcesses() throws Exception {
```

Comment:
yes. This is too low level - I think we will fix it at the higher levels. I dont' think deleteIndexDirectoryUnderLock should be lenient